### PR TITLE
Create table for shedlock

### PR DESCRIPTION
**Pull request by VithikaS**
### Checklist

- [ ] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

**Review by nscuro — COMMENTED** [review]
Looks good so far, glad the solution turned out to be very non-invasive and simple.

File: src/main/java/org/dependencytrack/tasks/metrics/PortfolioMetricsUpdateTask.java [hunk 1]
```diff
@@ -53,8 +67,24 @@ public class PortfolioMetricsUpdateTask implements Subscriber {
...
+            try (final QueryManager qm = new QueryManager()) {
+                PersistenceManager pm = qm.getPersistenceManager();
+                JDOPersistenceManagerFactory pmf = (JDOPersistenceManagerFactory) pm.getPersistenceManagerFactory();
+
+                JdbcLockProvider instance = new JdbcLockProvider(getDataSource(pmf));
+                LockingTaskExecutor executor = new DefaultLockingTaskExecutor(instance);
+                LockConfiguration lockConfiguration = new LockConfiguration(Instant.now(),
+                        PORTFOLIO_METRICS_TASK_LOCK.name(),
+                        Duration.ofMillis(Config.getInstance().getPropertyAsInt(TASK_PORTFOLIO_LOCK_AT_MOST_FOR)),
+                        Duration.ofMillis(Config.getInstance().getPropertyAsInt(TASK_PORTFOLIO_LOCK_AT_LEAST_FOR)));
```
[the remark this answers is not in the document]
Comment (VithikaS) [reply src/main/java/org/dependencytrack/tasks/metrics/PortfolioMetricsUpdateTask.java:79]: yes, we should. Connection was closing down in tests so I moved it like this for now. I will fix this

File: src/main/java/org/dependencytrack/tasks/LockName.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.dependencytrack.tasks;
+
+public enum LockName {
+    PORTFOLIO_METRICS_TASK_LOCK,
+    PROJECT_METRICS_TASK_LOCK,
```
Comment (mehab) [inline src/main/java/org/dependencytrack/tasks/LockName.java:5]: will PROJECT_METRICS_TASK_LOCK be used later on ?

Comment (VithikaS) [reply src/main/java/org/dependencytrack/tasks/LockName.java:5]: created for Project metrics task which could be used later on. We can remove it later if we don't want to lock it

**Review by mehab — APPROVED** [review]

**Pull request merged**